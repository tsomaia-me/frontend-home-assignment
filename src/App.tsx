import React, { useCallback, useEffect, useState } from 'react'
import { observer } from 'mobx-react'

import styles from './App.module.css'
import SearchField from './components/SearchField'
import PackageItem from './components/PackageItem/PackageItem'
import DependencyTree from './components/DependencyTree/DependencyTree'
import { usePackageStore } from './hooks/usePackageStore'
import Checkbox from './components/Checkbox'
import LoadingBurger from './components/LoadingBurger'

function App() {
  const packageStore = usePackageStore()
  const selectedPackage = packageStore.selectedPackage
  const [isAllExpanded, setIsAllExpanded] = useState(false)
  const [isHydrated, setIsHydrated] = useState(false)
  const toggleIsAllExpanded = useCallback(() => setIsAllExpanded(value => !value), [])

  useEffect(() => {
    packageStore.selectPackage(window.location.pathname.substring(1))
    setIsHydrated(true)
  }, [packageStore])

  return (
    <div className={styles.container}>
      <header className={styles.header}>
        NPM Package Database
      </header>

      <section className={styles.middle}>
        <aside className={styles.sidebar}>
          {isHydrated && <SearchField/>}
          <Checkbox
            label="Get entire tree"
            onClick={toggleIsAllExpanded}
          />
          <div className={styles.tree}>
            <DependencyTree
              dependencies={packageStore.topLevelDependencyList}
              isAllExpanded={isAllExpanded}
            />
          </div>
        </aside>

        <main className={styles.content}>
          {selectedPackage ? <PackageItem item={selectedPackage}/> : (
            <LoadingBurger isAnimationEnabled={packageStore.isTopLevelPackageLoading}/>
          )}
        </main>
      </section>
    </div>
  )
}

export default observer(App)
