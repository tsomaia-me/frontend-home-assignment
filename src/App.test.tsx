import React from 'react'
import renderer from 'react-test-renderer'
import App from './App'

test('renders App correctly', () => {
  const json = renderer.create(<App/>).toJSON()
  expect(json).toMatchSnapshot()
})
