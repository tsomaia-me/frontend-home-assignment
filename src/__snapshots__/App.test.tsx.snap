// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    NPM Package Database
  </header>
  <section
    className="middle"
  >
    <aside
      className="sidebar"
    >
      <label
        className="root"
      >
        <input
          onClick={[Function]}
          type="checkbox"
        />
        <span>
          Get entire tree
        </span>
      </label>
      <div
        className="tree"
      >
        <ul
          className="root"
        />
      </div>
    </aside>
    <main
      className="content"
    >
      <div
        className="root"
      >
        <span
          className="material-icons"
        >
          lunch_dining
        </span>
      </div>
    </main>
  </section>
</div>
`;
